newIDE/app/src/MainFrame/Toolbar/index.js

```diff
@@ -1,101 +1,92 @@
-              src="res/ribbon_default/export32.png"
-              disabled={!this.props.hasProject}
-              tooltip={t`Export the game (Web, Android, iOS...)`}
-            />
-          )}
-          {this.isDev && (
+          {isDev && props.showProjectIcons && <ToolbarSeparator />}
+          {isDev && (
             <ElementWithMenu
               element={<ToolbarIcon src="res/ribbon_default/bug32.png" />}
               buildMenuTemplate={() => [
                 {
                   label: 'Simulate update downloaded',
-                  disabled: !this.props.simulateUpdateDownloaded,
+                  disabled: !props.simulateUpdateDownloaded,
                   click: () => {
-                    this.props.simulateUpdateDownloaded &&
-                      this.props.simulateUpdateDownloaded();
+                    props.simulateUpdateDownloaded &&
+                      props.simulateUpdateDownloaded();
                   },
                 },
                 {
                   label: 'Simulate update available',
-                  disabled: !this.props.simulateUpdateAvailable,
+                  disabled: !props.simulateUpdateAvailable,
                   click: () => {
-                    this.props.simulateUpdateAvailable &&
-                      this.props.simulateUpdateAvailable();
+                    props.simulateUpdateAvailable &&
+                      props.simulateUpdateAvailable();
                   },
                 },
               ]}
             />
           )}
-          <ToolbarSeparator />
-          <PreviewButtons
-            onPreviewWithoutHotReload={this.props.onPreviewWithoutHotReload}
-            onOpenDebugger={this.props.onOpenDebugger}
-            onNetworkPreview={this.props.onNetworkPreview}
-            onHotReloadPreview={this.props.onHotReloadPreview}
-            setPreviewOverride={this.props.setPreviewOverride}
-            canDoNetworkPreview={this.props.canDoNetworkPreview}
-            isPreviewEnabled={this.props.isPreviewEnabled}
-            previewState={this.props.previewState}
-            hasPreviewsRunning={this.props.hasPreviewsRunning}
+        </ToolbarGroup>
+        <ToolbarGroup>
+          <PreviewAndPublishButtons
+            onPreviewWithoutHotReload={props.onPreviewWithoutHotReload}
+            onOpenDebugger={props.onOpenDebugger}
+            onNetworkPreview={props.onNetworkPreview}
+            onHotReloadPreview={props.onHotReloadPreview}
+            setPreviewOverride={props.setPreviewOverride}
+            canDoNetworkPreview={props.canDoNetworkPreview}
+            isPreviewEnabled={props.isPreviewEnabled}
+            previewState={props.previewState}
+            hasPreviewsRunning={props.hasPreviewsRunning}
+            exportProject={props.exportProject}
+            hasProject={props.hasProject}
           />
         </ToolbarGroup>
-        {this.state.editorToolbar || <ToolbarGroup />}
+        {editorToolbar || <ToolbarGroup />}
```

Comment:
You add a toolbarGroup to even the display? The `lastChild` is not useful here?

Reply:
hmm I've just tested without, it's not enough -> in order for the Preview buttons to be centered, it needs 2 Flex elements, one on left & one on right.